src/tests/system/tests/test_sssctl.py

```diff
@@ -242,287 +180,159 @@ def test_sssctl__check_missing_domain_name(client: Client):
-@pytest.mark.topology(KnownTopology.Client)
-def test_sssctl__check_misplaced_option(client: Client):
-    """
-    :title: sssctl config-check detects misplaced option
-    :setup:
-        1. In domain set "services" to "nss, pam"
-        2. Start SSSD, without config check
-    :steps:
-        1. Call sssctl config-check
-        2. Check error message
-    :expectedresults:
-        1. config-check detects an error in config
-        2. Error message is properly set
-    :customerscenario: False
-    """
-    client.sssd.common.local()
-    client.sssd.dom("test")["services"] = "nss, pam"
-
-    client.sssd.start(check_config=False)
-
-    result = client.sssctl.config_check()
-    assert result.rc != 0, "Config-check did not detect misconfigured config"
-
-    pattern = re.compile(r".Attribute 'services' is not allowed in section .*")
-    assert pattern.search(result.stdout), "Wrong error message was returned"
-
-
-@pytest.mark.tools
-@pytest.mark.topology(KnownTopology.Client)
-def test_sssctl__check_invalid_option_name_in_snippet(client: Client):
-    """
-    :title: sssctl config-check detects invalid option name in snippet
-    :setup:
-        1. Create new conf snippet with invalid option name
-    :steps:
-        1. Call sssctl config-check
-        2. Check error message
-    :expectedresults:
-        1. config-check detects an error in config snippet
-        2. Error message is properly set
-    :customerscenario: False
-    """
-    client.sssd.common.local()
-    client.fs.write("/etc/sssd/conf.d/01_snippet.conf", "[domain/local]\ninvalid_option = True", mode="600")
-
-    result = client.sssctl.config_check()
-    assert result.rc != 0, "Config-check did not detect misconfigured config snippet"
-    assert "Attribute 'invalid_option' is not allowed" in result.stdout, "Wrong error message was returned"
-
-
+@pytest.mark.parametrize(
+    "pattern,repl,expected",
+    [
+        pytest.param(
+            "id_provider.*",
+            "id_provider = invalid",
+            "Attribute 'id_provider' in section 'domain/local' has an invalid value: invalid",
+            marks=[pytest.mark.ticket(bz=2100789), pytest.mark.importance("high")],
+        ),
+        pytest.param(
+            "id_provider.*",
+            "",
+            "Attribute 'id_provider' is missing in section 'domain/local'.",
+            marks=[pytest.mark.ticket(bz=2100789), pytest.mark.importance("high")],
+        ),
+        ("id_provider", "id_@provider", "Attribute 'id_@provider' is not allowed in section"),
+        ("domain/", "d$main/", "Section [d$main/local] is not allowed"),
+        ("domain/local", "domain/local@", "Section [domain/local@] is not allowed"),
+        ("domain/local", "domainlocal", "Section [domainlocal] is not allowed"),
+        (".sssd.", "[sssdx]", "Section [sssdx] is not allowed"),
+        (".pam.", "[pamx]", "Section [pamx] is not allowed"),
+        (".nss.", "[nssx]", "Section [nssx] is not allowed"),
```

Comment:
I don't think so many repetitions of the same case - wrong section name - makes a lot of sense.

Reply:
I mean, there is no difference between '[sssdx]' vs '[pamx]' vs '[nssx]'

Reply:
This file hasn't gone through an audit. I assume because of the different output messages, that's why it was created. I'm all for removing the parametrization and reducing this to one test case if there is no difference.

Reply:
I would reduce 6 "section not allowed" to ~2 (1 with '/' and 1 without)